### PR TITLE
[DOCS] Document indices response parameters for node stats API

### PR DESCRIPTION
**Pull request by jrodewig**
- Documents `indices` parameters of the node stats API response body 
- Removes "flag" sentences from response body parm docs. These can be confusing and we explain how to pull metric groups elsewhere on the page.

Relates to #22199

### CODE REVIEW

File: docs/reference/cluster/nodes-stats.asciidoc [hunk 1]
```diff
@@ -138,6 +137,409 @@ include::{docdir}/rest-api/common-parms.asciidoc[tag=include-segment-file-sizes]
 [[cluster-nodes-stats-api-response-body]]
 ==== {api-response-body-title}
 
+[[cluster-nodes-stats-api-response-body-indices]]
+===== `indices` section
+
+The `indices` flag can be set 
+to retrieve information about indices:
```
Comment (jakelandis) [inline docs/reference/cluster/nodes-stats.asciidoc]: is this correct ?  `GET _nodes/stats` does not accept a `indices` param and this info is returned by default.  (or do i have the wrong API ?_

Comment (jrodewig) [reply docs/reference/cluster/nodes-stats.asciidoc]: The endpoint would be `_nodes/stats/indices`.

I borrowed the wording here from https://github.com/elastic/elasticsearch/pull/47525/files#diff-055d8b69ee66d0876311d63bc11fb382L144, but I can either remove it throughout. I feel like we explain how to pull particular metric groups well in the rest of the page.

Comment (jrodewig) [reply docs/reference/cluster/nodes-stats.asciidoc]: Removed this and similar sentences with 9b5c837.

Comment (jakelandis) [reply docs/reference/cluster/nodes-stats.asciidoc]: I don't have a strong opinion to include or not, I was thrown off by the term `flag` as I mentally mapped that to a `?query_param`.

File: docs/reference/cluster/nodes-stats.asciidoc [hunk 2]
```diff
@@ -138,11 +137,409 @@ include::{docdir}/rest-api/common-parms.asciidoc[tag=include-segment-file-sizes]
...
+`indices.docs.count`::
+(integer)
+include::{docdir}/rest-api/common-parms.asciidoc[tag=docs-count]
+
+`indices.docs.deleted`::
+(integer)
+include::{docdir}/rest-api/common-parms.asciidoc[tag=docs-deleted]
+
+`indices.store.size_in_bytes`::
+(integer)
```
Comment (jakelandis) [inline docs/reference/cluster/nodes-stats.asciidoc]: Do we differentiate between (Java) integer and (Java) long in the doc ? ...or is the just meant as whole number vs. fractional number thing ?

**Review by jakelandis — APPROVED** [review]
LGTM, one non-blocking question asking about integer vs. long

File: docs/reference/cluster/nodes-stats.asciidoc (hunk 2 above)
Comment (jrodewig) [reply docs/reference/cluster/nodes-stats.asciidoc]: It's meant as "whole number (integer) vs decimal number (float)".

That's consistent with how we've documented API reference so far, but I'm happy to make an exception here if needed. I could do something like `(64 bit signed integer)`.

I'd prefer not to refer folks to Java types if that's not strictly needed.

File: docs/reference/cluster/nodes-stats.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/cluster/nodes-stats.asciidoc]: Thanks! I think it works better without anyway.

**Pull request merged**